### PR TITLE
Update packaging guidelines

### PR DESCRIPTION
**Pull request by tmds**
This is an update of this document based on discussion with @leecow @omajid @livarcocc @JunTaoLuo @KathleenDollard @eerhardt @RheaAyase

### CODE REVIEW

**Review by BillWagner — APPROVED** [review]
This is a huge improvement.  Thanks @tmds 

I've reviewed the changes and I'm approving this.  @KathleenDollard do you want to review before we merge this?

**Comment by JRAlexander** [comment]
This is good stuff, @tmds!

File: docs/core/build/distribution-packaging.md
Comment (KathleenDollard) [inline docs/core/build/distribution-packaging.md]: Feels out of place under this heading. If we need to restate side-by-side this doesn't feel like the right heading, so I dropped that part and suggest a note if aside by side is important to the remainder of the discussion

"The host resides in a separate directory because there is only a single copy. This differs from other parts of the SDK which reside in individual directories. "

**Comment by leecow** [comment]
@BillWagner - this was heavily reviewed over in the designs repo and looks like it's good to do.

File: docs/core/build/distribution-packaging.md
Comment (tannergooding) [inline docs/core/build/distribution-packaging.md]: Should we add a sentence mentioning the target audience for this document For example: Users attempting to build from source or making changes that would impact building from source (such as the layout/etc).

File: docs/core/build/distribution-packaging.md
Comment (tannergooding) [inline docs/core/build/distribution-packaging.md]: Should we add a link to the roll-forward logic, so users can better understand when/why a particular runtime/sdk is selected.

File: docs/core/build/distribution-packaging.md
Comment (tannergooding) [inline docs/core/build/distribution-packaging.md]: Should we clarify that `dotnet restore` is not the only scenario. Such as `dotnet build /t:Restore` which also picks up this folder by default, as may other commands.

File: docs/core/build/distribution-packaging.md
Comment (tannergooding) [inline docs/core/build/distribution-packaging.md]: Do we need a link to somewhere that clarifies the distinction between .NET Core, .NET SDK, and .NET Runtime/etc and their version numbers?

File: docs/core/build/distribution-packaging.md
Comment (tannergooding) [inline docs/core/build/distribution-packaging.md]: OS package manager is Linux specific today. We don't have anything similar on Mac (unless a user is using something like Brew) or on Windows (unless a user is using something like chocolatey). However, in both cases, these are "unofficial" acquisition mechanism.

File: docs/core/build/distribution-packaging.md
Comment (tannergooding) [inline docs/core/build/distribution-packaging.md]: I don't have this folder (`shared/Microsoft.AspNetCore.All`) on my box. I also think this entire sentence is somewhat confusion and could use clarification.

File: docs/core/build/distribution-packaging.md
Comment (tannergooding) [inline docs/core/build/distribution-packaging.md]: This paragraph is also confusing. It could use clarification and possibly its own section (or it should possibly be added to one of the below sections, where it is more contextually relevant).

File: docs/core/build/distribution-packaging.md
Comment (tannergooding) [inline docs/core/build/distribution-packaging.md]: > These packages allows to install a specific patch version which is not automatically upgraded

Wording on this sentence is confusing... Posibly:
> These packages allows the user to install a specific patch version, which is not automatically upgraded

**Comment by tmds** [comment]
@tannergooding the document is intended for people packaging .NET Core, so mostly Linux maintainers. Focus during discussions was more on content than on wording. Since this is already merged, can you do a PR to improve?

**Comment by mairaw** [comment]
I feel link this change would warrant a new ms.date. Should I just check-in a new date for this @BillWagner @JRAlexander?

**Comment by BillWagner** [comment]
@mairaw Yes, this should have an updated date.

**Comment by tannergooding** [comment]
@KathleenDollard, are you interested in doing the new PR? I currently have a lot of other things I'm focusing on.

**Comment by mairaw** [comment]
I did open a new PR with some other fixes. I’ll take a look at your comments @tannergooding.

File: docs/core/build/distribution-packaging.md
Comment (mairaw) [reply docs/core/build/distribution-packaging.md]: Adding target audience is always desirable. Can you expand on what you mean by:

> making changes that would impact building from source (such as the layout/etc).

Comment (tannergooding) [reply docs/core/build/distribution-packaging.md]: I would imagine that the people who care about this document are:
* New users onboarding to the team
* Users attempting to build from source
* Users wanting to make changes to the CLI that could impact the resulting layout or packages produced (basically anyone making changes that would impact the things this document is spec'ing out)

File: docs/core/build/distribution-packaging.md
Comment (tmds) [reply docs/core/build/distribution-packaging.md]: @JunTaoLuo why is there no `shared/Microsoft.AspNetCore.All` folder? Is the text wrong?

Comment (tmds) [reply docs/core/build/distribution-packaging.md]: @tannergooding daily cli builds have the `Microsoft.AspNetCore.All` folder. So the text is ok I think.

**Pull request merged**
